packages/scripts/config/webpack.config.js

```diff
@@ -324,45 +387,46 @@ const config = {
 		! process.env.WP_NO_EXTERNALS &&
 			new DependencyExtractionWebpackPlugin(),
 	].filter( Boolean ),
-	stats: {
-		children: false,
-	},
 };
 
-// WP_DEVTOOL global variable controls how source maps are generated.
-// See: https://webpack.js.org/configuration/devtool/#devtool.
-if ( process.env.WP_DEVTOOL ) {
-	config.devtool = process.env.WP_DEVTOOL;
-}
+if ( hasExperimentalModulesFlag ) {
+	/** @type {webpack.Configuration} */
+	const moduleConfig = {
+		...baseConfig,
 
-if ( ! isProduction ) {
-	// Set default sourcemap mode if it wasn't set by WP_DEVTOOL.
-	config.devtool = config.devtool || 'source-map';
-	config.devServer = {
-		devMiddleware: {
-			writeToDisk: true,
-		},
-		allowedHosts: 'auto',
-		host: 'localhost',
-		port: 8887,
-		proxy: {
-			'/build': {
-				pathRewrite: {
-					'^/build': '',
-				},
+		entry: getWebpackEntryPoints( 'module' ),
+
+		output: {
+			...baseConfig.output,
+			module: true,
+			chunkFormat: 'module',
+			library: {
+				...baseConfig.output.library,
+				type: 'module',
 			},
 		},
+
+		plugins: [
+			new webpack.DefinePlugin( {
+				// Inject the `SCRIPT_DEBUG` global, used for development features flagging.
+				SCRIPT_DEBUG: ! isProduction,
+			} ),
+			new RenderPathsPlugin(),
```

Comment:
This plugin ensures PHP files referenced in block.json get copied. It ‘s probably fine to have it only in the scripts config.